### DISCUSSION TITLE
Suivi des utilisateurs grâce à un ID/cookie pour identifier la récurrence de la création de diagnostics

### DISCUSSION THREAD

**Discussion by NicolasM92** [MTES-MCT/sparte#400] [Idées]
Nous aimerions être capable d'identifier les utilisateurs qui reviennent régulièrement sur la plateforme pour créer des diagnostics.
Est-ce possible d'ajouter un ID sur les visiteurs pour pouvoir les identifier lorsqu'ils reviennent sur le site et pouvoir faire la distinction entre les premières utilisations et les utilisations fréquentes

**Comment by alexisig** [comment]
Est-ce que c'est pas déjà une information disponible avec le cookie `_gcl_au` de google analytics ?

**Comment by NicolasM92** [comment]
On l'a uniquement pour les personnes qui proviennent de Google Ads je crois


*Nicolas Mouanga*Chargé de déploiement Senior
***@***.***
06 66 22 88 96

**Comment by alexisig** [comment]
Je vois le cookie en accédant directement au site via l'url, mais il expire en 90 jours, donc pas adapté en effet.
Est-ce que tu peux préciser sur quels outils et de quel manière tu envisages de t'en servir ?

**Comment by NicolasM92** [comment]
On aimerait avoir l'info sur Metabase pour pouvoir calculer la part des
utilisateurs qui viennent plus de x fois sur le site et pouvoir calculer
nos taux de rétention à terme.


*Nicolas Mouanga*Chargé de déploiement Senior
***@***.***
06 66 22 88 96

**Reply by NicolasM92** [reply]
@alexisig est-ce que le point du dessus précise assez le besoin ?
Pour essayer de préciser davantage, la finalité pour nous serait de pouvoir grouper les typologies d'action pour chaque utilisateur.
Je ne sais pas vraiment comment ça devrait se décliner dans Metabase, peut-être l'ajout d'une colonne "ID" qui nous permettrait ensuite de grouper nos différentes requêtes par ID ?

**Reply by alexisig** [reply]
On a plusieurs possibilités pour essayer de mettre ça en place, mais aucune ne permettra réellement d'avoir des statistiques fiables sur les utilisateurs anonymes. Cela dit ça permettrait d'avoir des tendances, par exemple : on pourrait savoir si un utilisateur est revenu, mais on ne pourrait pas dire avec certitude combien de fois.

La limitation se situe au niveau de l'utilisation de cookie : ils peuvent être supprimés par les utilisateurs, ils expirent (pour des raisons de sécurité) et ils ne suivent pas les utilisateurs d'un device à un autre (ordinateur à mobile, ou ordinateur de travail à ordinateur perso, etc.).

Django utilise des cookies pour stocker les sessions. Ce qu'on peut faire c'est envoyer l'id de ces session à métabase. C'est built-in à Django donc c'est pas une énorme tâche pour nous, mais il faut qu'on soit conscient des limitations, ça ne permettra pas de faire ce qu'on pourrait faire avec des vrais comptes utilisateurs.
